extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/execution/SelectorBasedPodTemplateSelectStrategy.java

```diff
@@ -64,9 +60,9 @@ public PodTemplate getPodTemplateForTask(Task task, Map<String, PodTemplate> tem
                                   .filter(selector -> selector.evaluate(task))
                                   .findFirst()
                                   .map(Selector::getSelectionKey)
-                                  .orElse(defaultKey);
+                                  .orElse("base");
 
-    return templates.getOrDefault(templateKey, templates.get("base"));
```

Comment:
This change is not right, there is a matched template key doesn't mean the template exist, fallback to `base` template key doesn't solve this problem.